main.rb

```diff
@@ -1,49 +1,13 @@
 require_relative 'app'
-
-def all_options
-  puts "\nPlease choose an option by enterin a number:\n
-1 - List all Books\n
-2 - List all People\n
-3 - Create a Person\n
-4 - Create a Book\n
-5 - Create a Rental\n
-6 - List all rentals for a given person id\n
-7 - Exit"
-end
-
-def prompt
-  puts 'Welcome to School Library App!'
-  loop do
-    all_options
-    option = gets.chomp.to_i
-    break if option == 7
-
-    call_option(option)
-  end
-end
-
-def call_option(option)
-  case option
-  when 1
-    list_books
-  when 2
-    list_people
-  when 3
-    create_person
-  when 4
-    create_book
-  when 5
-    create_rental
-  when 6
-    list_rentals
-  else
-    puts 'Choose between 1 - 7'
-  end
-end
+require_relative 'book'
+require_relative 'student'
+require_relative 'rental'
+require_relative 'teacher'
+require_relative 'menu'
 
 def main
   app = App.new
-  app.run
+  run_menu(app)
 end
```

Comment:
- [ ] Well done :+1: you implemented the changes but your app crashes when I run it so kindly fix it check below :point_down: the error I got and fix it :slightly_smiling_face: 

![Screenshot from 2024-01-11 22-39-06](https://github.com/PrinceBadru/OOP-school-library/assets/45226848/ac13230d-ad57-4aac-a0b1-51261c68e381)